tests/robot-tests/scripts/create_snapshots.py

```diff
@@ -34,21 +34,31 @@ def create_find_statistics_snapshot(public_url) -> str:
             'topics': [],
         }
 
-        topics = theme_html.select('details') or []
+        topics = theme_html.select('[id^="topic-details-"]') or []
         for topic_html in topics:
             topic = {
                 'topic_heading': topic_html.select_one('[id^="topic-heading-"]').string,
-                'publications': [],
+                'publication_types': [],
             }
 
-            publications = topic_html.select('li') or []
-            for publication_html in publications:
-                publication = {
-                    'publication_heading': publication_html.select_one('[id^="publication-heading-"]').string,
-                    'on_ees': publication_html.find(text="View statistics and data") is not None,
+            publication_types = topic_html.select('[id^="publication-type-heading-"]') or []
+            for publication_type_html in publication_types:
+                publication_type_key = publication_type_html['id'].replace('publication-type-heading-', '')
+                publication_type = {
+                    'publication_type_heading': publication_type_html.string,
+                    'publications': [],
                 }
 
-                topic['publications'].append(publication)
+                publications = topic_html.select(
+                    f'ul[data-testid="publications-list-{publication_type_key}"] > li') or []
```

Comment:
I was expecting you to use `publication_type_html` here - then potentially you can avoid all the `publication_type_key` stuff. Maybe it doesn't work?

Reply:
Personally, I'm not against wrapping stuff with a div with `data-testid` where necessary. If it makes it easier, go ahead, but of course, avoid it if there are better alternatives :smile:

I'd also say that general preference is towards `data-testid` instead of `id` as we can avoid potential screen reader issues from having multiple elements with the same id. It does make sense to use `id` for headings as it's fairly common to see this on other websites (e.g. in documentation) so that you can easily navigate back to a section by putting the id in the URL hash.

Reply:
Thanks, for some reason I thought it would be bad to that. It certainly makes things a lot easier here in the snapshot script where the code now follows the same convention, and also in the Jest tests.